### ISSUE TITLE
What is the most stable storage for conductor?

### ISSUE THREAD

**Issue by andyyumiao** [Netflix/conductor#2059]
As title, what is the most stable storage for conductor? Cassandra、Redis、Mysql or PostgreSQL? I mean the recommended storage storage...

**Comment by apanicker-nflx** [comment]
The choice of storage depends on your use-case and how distributed you want your Conductor deployment to be. 
At Netflix, we are currently using Cassandra as our primary datastore. We use an internal queuing system built on top of Redis for the queuing layer. Further, we also Redis for the rate limiting dao implementation. This is supported by Elasticsearch (v6) which acts as the indexing layer (secondary datastore). Few of these were chosen for legacy reasons and to fulfill niche requirements(YMMV). 

One of the advantages of using Redis, MySQL or PostgreSQL is that these implementations provide support for both persistence and queuing layers. We have an active community of users using the MySQL and PostgreSQL implementations in their production environments.

**Comment by andyyumiao** [comment]
> The choice of storage depends on your use-case and how distributed you want your Conductor deployment to be.
> At Netflix, we are currently using Cassandra as our primary datastore. We use an internal queuing system built on top of Redis for the queuing layer. Further, we also Redis for the rate limiting dao implementation. This is supported by Elasticsearch (v6) which acts as the indexing layer (secondary datastore). Few of these were chosen for legacy reasons and to fulfill niche requirements(YMMV).
> 
> One of the advantages of using Redis, MySQL or PostgreSQL is that these implementations provide support for both persistence and queuing layers. We have an active community of users using the MySQL and PostgreSQL implementations in their production environments.

yes, we have used Mysql as the storage in our production environment, it's also stable enough.

**Comment by hi-zjh** [comment]
> > The choice of storage depends on your use-case and how distributed you want your Conductor deployment to be.
> > At Netflix, we are currently using Cassandra as our primary datastore. We use an internal queuing system built on top of Redis for the queuing layer. Further, we also Redis for the rate limiting dao implementation. This is supported by Elasticsearch (v6) which acts as the indexing layer (secondary datastore). Few of these were chosen for legacy reasons and to fulfill niche requirements(YMMV).
> > One of the advantages of using Redis, MySQL or PostgreSQL is that these implementations provide support for both persistence and queuing layers. We have an active community of users using the MySQL and PostgreSQL implementations in their production environments.
> 
> yes, we have used Mysql as the storage in our production environment, it's also stable enough.

i use mysql too,but the task is always in scheduled status, server log is "No Task available for the poll"

i use 3.0.1 version, here is my config ,can you help me find the problem?
`conductor.db.type=mysql
conductor.mysql.jdbcUrl=jdbc:mysql://10.101.1.217:9672/conductor
conductor.mysql.jdbcUsername=rw_conductor
conductor.mysql.jdbcPassword=rjYdXx76HU8A4JUq
# Hikari pool sizes are -1 by default and prevent startup
conductor.mysql.connectionPoolMaxSize=50
conductor.mysql.connectionPoolMinIdle=5
conductor.mysql.flywayEnabled=false

conductor.indexing.enabled=true

conductor.workflow-repair-service.enabled=true

conductor.elasticsearch.url=http://10.101.129.202:11445
conductor.elasticsearch.indexName=conductor
conductor.elasticsearch.version=6


conductor.app.workflow-execution-lock-enabled=false
conductor.workflow-execution-lock.type=noop_lock
`

**Comment by ab48917** [comment]
@jeff-jh did you get the working queue working using redis? or just using mysql as persistance with conductor?

**Comment by jeff-jh** [comment]
> @jeff-jh did you get the working queue working using redis? or just using mysql as persistance with conductor?

i just using mysql as persistance and  working queue with conductor. i have solved this problem by update conductor version

**Comment by ab48917** [comment]
basically you are using conductor.db.type=mysql i.e. only mysql and no redis
